src/taipy/gui/gui.py

```diff
@@ -1238,7 +1237,7 @@ def __get_on_cancel_block_ui(self, callback: t.Optional[str]):
         def _taipy_on_cancel_block_ui(guiApp, id: t.Optional[str], payload: t.Any):
             if _hasscopeattr(self, Gui.__UI_BLOCK_NAME):
                 _setscopeattr(self, Gui.__UI_BLOCK_NAME, False)
-            self.__on_action(id, callback)
+            self.__on_action(id, {"action": callback})
```

Comment:
interesting: should self be guiApp here ?

Reply:
Hmmmm I guess you're right!
Lucky we are that no one is creating multiple Gui instances hu?